### DISCUSSION TITLE
Lab3 result data length

### DISCUSSION THREAD

**Discussion by DanielDu9453** [bol-edu/HLS-SOC-Discussions#94] [Q&A]
想請問如果是做FIR，結果的長度應該是input data length + filter length -1，但是提供的golden answer只有input data length筆，那我們是不需要算後面的結果嗎?還是也是要計算並輸出

**Comment by kevin861222** [comment]
Testbech的驗證是從第一筆x輸入開始計算（其他10個data當成0）輸出第1筆Y，接著第二筆x輸入，（其他9個data當成0）輸出第2筆Y，直到最後一筆x輸入，輸出第input data length筆Y，你按照這樣的方式設計就能通過TB。
但的確如你所說，這樣的計算不是標準的FIR。

**Reply by DanielDu9453** [reply]
對 所以我是想問助教輸出需不需要有input data length + 10筆

**Reply by kevin861222** [reply]
了解
